src/Stratis.Bitcoin.Features.WatchOnlyWallet/WatchOnlyWalletManager.cs

```diff
@@ -155,11 +231,75 @@ public void ProcessTransaction(Transaction transaction, Block block = null)
                         {
                             existingTransaction.MerkleProof = new MerkleBlock(block, new[] { transactionHash }).PartialMerkleTree;
                         }
+
+                        // Update the lookup cache with the new transaction information.
+                        this.txLookup.Remove(existingTransaction.Id);
+                        this.txLookup.Add(existingTransaction.Id, existingTransaction);
                     }
 
                     this.SaveWatchOnlyWallet();
                 }
             }
+
+            this.Wallet.WatchedTransactions.TryGetValue(transactionHash.ToString(), out TransactionData existingWatchedTransaction);
+
+            if (existingWatchedTransaction != null)
+            {
+                // The transaction was previously stored, in an unconfirmed state.
+                // So now update the block hash and Merkle proof since it has
+                // appeared in a block.
+                existingWatchedTransaction.BlockHash = block.GetHash();
```

Comment:
I think it may be possible that `block` will be `null` here

Reply:
No its fine see comment above `The transaction was previously stored, in an unconfirmed state.` if its confirmed there should be a block.

Reply:
Thanks, I've missed that.

Reply:
Actually, I think you're right. It is possible for ProcessTransaction to be called with the block parameter set to null.